### PR TITLE
new optimization method

### PR DESCRIPTION
**Pull request by LousyDuck**

### CODE REVIEW

File: tracerj/src/main/java/calculus/functions/MathFunction.java [hunk 1]
```diff
@@ -22,14 +23,20 @@ public double integrate(double from, double to, int slices) {
         return Integrals.simpsonsRule(this::applyAsDouble, from, to, slices);
     }
 
-    public double pointAtLength(double start, double length, double accuracy) {
-        double sum = 0;
-        double x = start;
+    public double binarySearchPercentageAtLength(MathFunction lengthFunctionDerivative, double start, double length, double maxPercentage, double accuracy) {
```
Comment (tomtzook) [inline tracerj/src/main/java/calculus/functions/MathFunction.java]: Extract to a different class whose responsibility will be to calculate the point at length or something

File: tracerj/src/main/java/calculus/splines/parameters/SplineParameters.java [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
     private final Waypoint end;
     private final double startLength;
+    private final double maxDistancePassedPerCycle;
 
-    public SplineParameters(Waypoint start, Waypoint end, double startLength) {
+    public SplineParameters(Waypoint start, Waypoint end, double startLength, double maxDistancePassedPerCycle) {
         this.start = start;
         this.end = end;
         this.startLength = startLength;
+        this.maxDistancePassedPerCycle = maxDistancePassedPerCycle;
```
[the remark this answers is not in the document]
Comment (LousyDuck) [reply tracerj/src/main/java/calculus/splines/parameters/SplineParameters.java]: not a thing lol

File: tracerj/src/main/java/calculus/functions/MathFunction.java (hunk 1 above)
Comment (LousyDuck) [reply tracerj/src/main/java/calculus/functions/MathFunction.java]: problematic

**Comment by LousyDuck** [comment]
please review or approve asap because I have to back merge many changes that are made on dev to prevent conflicts ;-;

**Pull request closed without merging**